accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-input-rules.ts

```diff
@@ -143,6 +144,23 @@ let a11yRulesInput: Rule[] = [
                 passed = RPTUtil.attributeNonEmpty(ruleContext, "label") || ruleContext.innerHTML.trim().length > 0;
                 if (!passed) POF = 2 + textTypes.length + buttonTypes.length + 3;
             }
+            if (!passed) {
+                // check aria role
+                //TODO: consider other aria roles relevant, other than menuitemcheckbox
+                const singleRole = RPTUtil.hasRoleInSemantics(ruleContext, "menuitemcheckbox") || RPTUtil.hasRoleInSemantics(ruleContext, "menuitemradio")
+                                || RPTUtil.hasRole(ruleContext, "radio", false);
+                const otherRole = RPTUtil.hasRole(ruleContext, "listbox", false) || RPTUtil.hasRole(ruleContext, "textbox", false) 
+                                || RPTUtil.hasRole(ruleContext, "searchbox", false);
+                                
+                if (singleRole)
+                    passed = RPTUtil.getInnerText(ruleContext) && RPTUtil.getInnerText(ruleContext).trim().length > 0;
+                else if (otherRole) {
+                    passed = RPTUtil.attributeNonEmpty(ruleContext, "aria-label") || RPTUtil.attributeNonEmpty(ruleContext, "aria-labelledby")
+                             || RPTUtil.hasNonEmptyTextNode(ruleContext);
```

Comment:
Sorry - should have looked at this more closely. `hasNonEmptyTextNode` is unnecessary since none of these roles (`listbox`, `textbox`, `searchbox`) are labelled by their children.

Reply:
this is for the element that can have its own text, for example, <div role='listbox'>My Div</div>, it should be equivalent to <div id='labeldiv''>My Div<div role='listbox' aria-labelledby='labeldiv'></div>.  In this case, the text content is not from it's innerText or textContent, which includes text from all the children.

Reply:
fixed the name calculation.